cfe_internal/update/update_policy.cf

```diff
@@ -299,14 +297,112 @@ bundle agent cfe_internal_update_policy_cpv
+      "_cmdb_previous_next_request_from"
+        string => readfile( $(_cmdb_next_request_state_file), inf ),
+        if => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      "_cmdb_previous_next_request_from"
+        string => "0",
+        unless => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      # We need a script to call that should return the API response
+      "_get_cmdb_data_bin" string => "$(sys.bindir)/get_cmdb.php";
+      "_get_cmdb_data_cmd" string => "/var/cfengine/httpd/php/bin/php $(_get_cmdb_data_bin) $(_cmdb_previous_next_request_from)";
+
+      # We call the script and we pass it the timestamp from the prior call
+      "_get_cmdb_data_response"
+        string => execresult( $(_get_cmdb_data_cmd), useshell ), # This would be execresult()
+        if => fileexists( $(_get_cmdb_data_bin) );
+
+      "_get_cmdb_data_response_d"
+        data => mergedata('$(_get_cmdb_data_response)');
+
+      # So that we can write a JSON file for each host we get the indicies of data in the response
+      "_i" slist => getindices( "_get_cmdb_data_response_d[data]");
+
+      # We need to store the timestamp from the most recent change so that we can use that as a starting point for future requests.
+      "_next_request_from"
+        string => "$(_get_cmdb_data_response_d[meta][cmdb_epoch])";
+
+  files:
+      # "$(_get_cmdb_data_cmd)" perms => m( 700 );
+
+@if minimum_version(3.18)
+      # This functionality is only present on 3.18.0+ Enterprise hubs, and this
+      # promise uses the /content/ attribute which was first introduced in
+      # 3.16.0.
+
+      # If the next request state file doesn't exist, we seed one with 0, the
+      # lowest epoch value possible. because we populate variables from this
+      # file content.
+      "$(_cmdb_next_request_state_file)"
+        content => "0$(const.n)",
+        handle => "cmdb_data_change_next_seed",
+        unless => fileexists("$(_cmdb_next_request_state_file)" );
+@endif
+
+      # Write out the data for each host that had a data change
+      "$(sys.workdir)/cmdb/$(_i)/host_specific.json"
+        create => "true", # CFE-2329, ENT-4792
+        template_data => mergedata("_get_cmdb_data_response_d[data][$(_i)]" ),
+        template_method => "inline_mustache",
+        edit_template_string => string_mustache( "{{$-top-}}", "_get_cmdb_data_response_d[data][$(_i)]" ),
+        if => isgreaterthan( $(_next_request_from), $(_cmdb_previous_next_request_from) );
+
+      # Write out the last data change timestamp so we can use it as a startring point
+      "$(_cmdb_next_request_state_file)"
+        handle => "cmdb_data_change_next_update",
+        template_data => parsejson('{}'),
+        template_method => "inline_mustache",
+        edit_template_string => "$(_next_request_from)",
+        if => isgreaterthan( $(_next_request_from), $(_cmdb_previous_next_request_from) );
```

Comment:
Could also use `content => "$(_next_request_from)$(const.n)"`

Reply:
Yeah

Reply:
I did this, as you wish.